lib/vcs.ml

```diff
@@ -64,28 +64,49 @@ let find_git () =
       | dir, (_, `Exited 0) -> Ok (Some (v `Git git ~dir:Fpath.(v dir)))
       | _ -> Ok None )
 
-let cmd_error cmd = function
-  | `Exited c ->
-      R.error_msgf "The following command exited with code %d:@\n%a" c Cmd.pp
-        cmd
-  | `Signaled c ->
-      R.error_msgf "The following command exited with signal %d:@\n%a" c Cmd.pp
-        cmd
+let cmd_error cmd ?err_msg =
```

Comment:
What would you think about moving this to `Sos` instead? I think it could benefit to more places in the code than just `Vcs` and it seems generic enough!

Also one thing I'm wondering is why is `err_msg` optional? Shouldn't it be mandatory? I feel like if one wants to report a command failure they should have run it with `run_out_err` and should always pass this function the command's stderr.

Reply:
Ok, I've moved it to `Sos` . 

The reason why `err_msg` was optional was because there are still commands for which I've consciously decided not to print out the error message from the external command (`git_file_is_dirty` and `run_hg`). As you've suggested on Slack, I've now made the parameter `err_msg` mandatory (but being an option type), so that we can't just forget to pass the error message to the function. I think that's a good idea.